db/compaction/compaction_picker_universal.cc

```diff
@@ -373,11 +390,17 @@ Compaction* UniversalCompactionBuilder::PickCompaction() {
       "[%s] Universal: sorted runs files(%" ROCKSDB_PRIszt "): %s\n",
       cf_name_.c_str(), sorted_runs_.size(), vstorage_->LevelSummary(&tmp));
 
-  // Check for size amplification first.
   Compaction* c = nullptr;
-  if (sorted_runs_.size() >=
-      static_cast<size_t>(
-          mutable_cf_options_.level0_file_num_compaction_trigger)) {
+  if (!vstorage_->FilesMarkedForPeriodicCompaction().empty()) {
```

Comment:
Does periodic compaction have higher priority than size-amp-based compaction?

Reply:
I think it should be the case, as periodic compaction is a hard requirement. Is it the case in the code?